kn/fotos/entities.py

```diff
@@ -248,6 +257,15 @@ def update_metadata(self, parent, save=True):
     def original_path(self):
         return os.path.join(settings.PHOTOS_DIR, self.path, self.name)
 
+    @property
+    def mongo_path_prefix(self):
+        if self.path is None:
+            # root
+            return {'$exists': True, '$ne': None}
+        else:
+            return {'$regex': re.compile(
+                                "^%s(/|$)" % re.escape(self.full_path))}
+
```

Comment:
Bij de root krijg je de regex `^(/|$)` want `full_path` is een lege string. Dus worden alleen `path`s die helemaal leeg zijn of beginnen met een slash gematcht, dus alleen de root zelf.

Ik kon zo gauw geen regex bedenken die wél rekening houdt met de root.